### DISCUSSION TITLE
关于搬砖功能

### DISCUSSION THREAD

**Discussion by wnykuang** [ccagml/leetcode-extension#294] [Ideas]
你可以看看anki 相关的记忆曲线的内容

我自己就是用一个anki 的deck 来记录各种题目的

顺便，你要是有兴趣

**Comment by ccagml** [comment]
你是怎么记录的呢?
搬砖 这个功能 一开始感觉根据遗忘曲线的原理一样定时回顾题目可能会有帮助
实际上我用起来好像没啥用

**Reply by wnykuang** [reply]
> 你是怎么记录的呢?
> 
> 搬砖 这个功能 一开始感觉根据遗忘曲线的原理一样定时回顾题目可能会有帮助
> 
> 实际上我用起来好像没啥用

我自己算是算法苦手

在anki 里
我会丢到某个deck（like：经典题）
然后给题目打tag
卡片正面写题目链接

然后想复习的时候建一个filter
“deck：经典题 tag：双指针” 然后选择relative overdueness ，限制自己复习3-5 题（太多会太花时间）

**Reply by ccagml** [reply]
这样理解,deck相当于现在搬砖的一个自定义分类, tag是题目的tag, 随机的范围从这里面选?
这样理解对吗?

**Reply by wnykuang** [reply]
> 这样理解,deck相当于现在搬砖的一个自定义分类, tag是题目的tag, 随机的范围从这里面选?
> 
> 这样理解对吗?

差不多
如果究极简化，不太需要deck
能控制题目的tag（如果题目自己有tag 其实也不需要tag）


说起来这个vscode 插件你是怎么调试的？我没怎么写过前端，看了看npm 相关的那几个选项好像也没调试的了

**Reply by ccagml** [reply]
.vscode目录配置的launch.json就是调试配置了

**Comment by p59740** [comment]
算法苦手+1, 
我以前也用 Anki 来记, 但感觉想存下来自己的笔记和注意点, anki 始终比较零散不太友好, 后来我在 notion 里搭建了一个模板, 给每个笔记设置了些属性; 按功能做了些view, 比如[今日复习到期View ](https://helpful-horse-84b.notion.site/Review-Today-f7976352a3f5420eb60d45505c336f80),[ data Structure View ](https://helpful-horse-84b.notion.site/DataStructure-cb5ee1c9c7ea426397980116b88df00f) )

现在在 notion 里, 借助 anki 的概念, 使用上有 2 种状态, 学习"is learning"  和 复习"is reviewing" : 
1. 当天没有 AC 的题(看题解也算), 状态 = "is learning", 第二天要继续写,继续学习, 直到学会了进入复习"is reviewing"; 
2. AC 了的题, 状态 = "is reviewing" ,按艾宾浩斯曲线安排复习; 
3. AC 了的题在到期复习时, 但不会写了, 允许重置为 学习"is learning"  , 即转换成第 1 类, 会在次日重新学.

@ccagml 大佬的插件相见恨晚, 要是早点用上我也许就不用搞 notion 的模板了哈哈. 上面的思路给 搬砖功能  1 个参考~

-----------------
Btw, 这个插件真的是最好用的 leetcode 插件!!! 感恩!!! (第 231 个 star, 主页描述可以改掉 230 了 /doge)
